packages/next/src/build/utils.ts

```diff
@@ -1573,9 +1582,7 @@ export function detectConflictingPaths(
         conflictingPage = dynamicSsgPages.find((page) => {
           if (page === pathsPage) return false
 
-          conflictingPath = additionalSsgPaths
-            .get(page)
-            ?.find((compPath) => compPath.toLowerCase() === lowerPath)
```

Comment:
this `?.find(...)...` is an array search _within_ an array search, in sites with hundreds of thousands of paths, this becomes a huge problem.